### PR TITLE
84 learn front end

### PR DESCRIPTION
**Pull request by ajross4**
Covering VisQ.AI changes since 8/22/2025...

## Summary by CodeRabbit

* **New Features**
  * Built-in database migration system with versioned migrations, backups, rollback, integrity validation, and migration history.

* **Improvements**
  * Smarter file-selection dialog with working-directory validation and file preselection.
  * Import progress shows real progress, supports cancellation, and keeps the UI responsive.
  * Model-path handling can bind to parent-provided handlers when available.
  * Tab navigation no longer auto-advances when clicking the current or previous step.
  * Predictor initialization disables legacy Keras to improve model loading.

* **Bug Fixes**
  * Prevents invalid paths and improves robustness of import and temperature interpolation.
  * Removed automatic run-directory prefill where inappropriate.

* **Tests**
  * Comprehensive tests for migrations, backups, rollbacks, integrity, edge cases, and concurrency.

### CODE REVIEW

File: QATCH/VisQAI/src/view/main_window.py [hunk 1]
```diff
@@ -368,9 +368,13 @@ def eventFilter(self, obj, event):
...
-                if hasattr(self.tab_widget.currentWidget(), "btn_next"):
-                    self.tab_widget.currentWidget().btn_next.click()
+                if now_step >= tab_step:
+                    # do not click next when user is going backwards
+                    # (or nowhere) from the currently selected step.
+                    pass
+                elif hasattr(self.tab_widget.currentWidget(), "btn_next"):
                     # still perform click action
+                    self.tab_widget.currentWidget().btn_next.click()
         return super().eventFilter(obj, event)
```
[the remark this answers is not in the document]
Comment (ajross4) [reply QATCH/VisQAI/src/view/main_window.py:378]: We actually do still want that tab click event to fire, as shown on line 378. The `btn_next.click()` call is included to save any state from the active tab before changing to the new tab that the user has clicked on (by **not** blocking the action). Doing as suggested would always move the tab widget forward one tab, and not adhere to the user's wishes.

File: QATCH/VisQAI/src/view/frame_step1.py [hunk 2]
```diff
@@ -1272,6 +1301,14 @@ def file_selected(self, path: str | None, cancel: bool = False):
...
         self.btn_update.setEnabled(False)
 
+        if Constants.log_prefer_path not in path:
+            Log.e("The selected run is not in your working directory and cannot be used.")
+            Log.e("If desired, please change your working directory to use this run.")
+            # deselect file path, will show log_prefer_path next time
+            path = None
+            self.run_file_run = None
+            # NOTE: method will return after next block
+
```
[the remark this answers is not in the document]
Comment (ajross4) [reply QATCH/VisQAI/src/view/frame_step1.py:1311]: Fixing in next commit
